rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -256,6 +254,17 @@ NodeBase::get_callback_groups() const
   return callback_groups_;
 }
 
+void NodeBase::callback_groups_for_each(const CallbackGroupCallback & func)
+{
+  std::lock_guard<std::mutex> lock(callback_groups_mutex_);
+  for (rclcpp::CallbackGroup::WeakPtr & weak_group : this->callback_groups_) {
+    rclcpp::CallbackGroup::SharedPtr cur_group = weak_group.lock();
+    if (cur_group) {
+      func(cur_group);
+    }
+  }
```

Comment:
nitpick: I always suggest avoiding any kind of abbreviation because it makes reading code harder for people who don't speak english, i.e. `cur_group` probably doesn't translate automatically very well. So either `current_group` or just `group`:

```suggestion
  for (rclcpp::CallbackGroup::WeakPtr & weak_group : this->callback_groups_) {
    rclcpp::CallbackGroup::SharedPtr group = weak_group.lock();
    if (group) {
      func(group);
    }
  }
```

I actually feel the same way about `func` and would use `function` myself, but that one is common enough that it probably won't confuse anyone.

Reply:
Yeah, makes sense.  This was actually just copy-n-pasted from `callback_group_in_node` above, but I'll change it to just "group".  I'm going to leave "func" as-is, since that one is pretty clear.